### PR TITLE
fix(shield): add default host shield tolerations and affinity

### PR DESCRIPTION
**Pull request by aroberts87**
## What this PR does / why we need it:
The Shield chart was not defining a default set of node affinity and toleration settings which resulted in the Host Shield not being installed on expected nodes.

## Checklist

- [x] Title of the PR starts with type and scope, (e.g. `feat(agent,node-analyzer,sysdig-deploy):`)
- [x] Chart Version bumped for the respective charts
- [x] Check GithubAction checks (like lint) to avoid merge-check stoppers
- [x] All test files are added in the tests folder of their respective chart and have a "_test" suffix

### CODE REVIEW

File: charts/shield/templates/host/_helpers.tpl [hunk 1]
```diff
@@ -54,24 +54,44 @@ If release name contains chart name it will be used as a full name.
...
 {{- end -}}
 
 {{- define "host.affinity" -}}
 {{- $affinity := merge (dict) .Values.affinity .Values.host.affinity -}}
+{{- if $affinity }}
 {{- with $affinity -}}
 {{- . | toYaml -}}
 {{- end -}}
+{{- else -}}
+nodeAffinity:
```
Comment (AlbertoBarba) [inline charts/shield/templates/host/_helpers.tpl]: Why don't we set this as default in the values for `host.affinity`?

File: charts/shield/values.yaml [hunk 2]
```diff
@@ -206,7 +206,23 @@ host:
   # The node selector for the host shield
   node_selector: {}
   # The tolerations for the host shield
-  tolerations: []
+  tolerations:
```
Comment (AlbertoBarba) [inline charts/shield/values.yaml]:
```suggestion
  tolerations: # +doc-gen:break
```
And re-run `make docs`

**Review by AlbertoBarba — CHANGES_REQUESTED** [review]

File: charts/shield/templates/host/_helpers.tpl (hunk 1 above)
Comment (aroberts87) [reply charts/shield/templates/host/_helpers.tpl]: There wasn't a strong reason not to. I was just exploring an option here to not clutter up the default `values.yaml` with verbose default settings. I'm open to moving this out to the file though. What are your thoughts?

Comment (AlbertoBarba) [reply charts/shield/templates/host/_helpers.tpl]: The only good point i have in my mind for having it in the values it that the generated documentation will report it as default value, while doing it in this way will hide it in the helpers

Comment (aroberts87) [reply charts/shield/templates/host/_helpers.tpl]: That's a good point. 🤔 👍🏻

Comment (mavimo) [reply charts/shield/templates/host/_helpers.tpl]: I agree on hiding from the doc (but maybe we may have some annotation for that?) but adding in the default values looks like more clear to users, plus will make the code simpler here  (no if else, no special scenario, ..) 🤔

**Review by AlbertoBarba — APPROVED** [review]

**Pull request merged**